packages/@uppy/compressor/src/index.js

```diff
@@ -52,12 +52,12 @@ export default class Compressor extends BasePlugin {
           this.uppy.log(`[Image Compressor] Image ${file.id} compressed by ${prettierBytes(compressedSavingsSize)}`)
           totalCompressedSize += compressedSavingsSize
           const { name, type, size } = compressedBlob
-          const { extension } = getFileNameAndExtension(name)
+          const extension = name && getFileNameAndExtension(name).extension
           this.uppy.setFileState(file.id, {
-            name,
-            extension,
-            type,
-            size,
+            ...(name && { name }),
+            ...(extension && { extension }),
+            ...(type && { type }),
+            ...(size && { size }),
```

Comment:
Looks goofy but it works

Reply:
Yeah, I agree, do you have a suggestion?